x-pack/test/security_solution_cypress/cypress/e2e/detection_response/rule_management/prebuilt_rules/upgrade_with_preview.cy.ts

```diff
@@ -0,0 +1,1638 @@
+          newRuleAssets: [NEW_PREBUILT_RULE_ASSET],
+        });
+        visitRulesUpgradeTable();
+        openPrebuiltRuleUpgradeFlyoutFor('Customized prebuilt rule');
+
+        cy.get(UPDATE_PREBUILT_RULE_BUTTON).should('be.disabled');
+
+        saveFieldValue('name');
+
+        cy.get(UPDATE_PREBUILT_RULE_BUTTON).should('be.enabled');
+      });
+
+      it('is disabled when num fields in edit mode >= 1', () => {
+        const PREBUILT_RULE_ID = 'test-prebuilt-rule';
+        const PREBUILT_RULE_ASSET = createRuleAssetSavedObject({
+          rule_id: PREBUILT_RULE_ID,
+          version: 2,
+          name: 'Test prebuilt rule',
+        });
+        const NEW_PREBUILT_RULE_ASSET = createRuleAssetSavedObject({
+          rule_id: PREBUILT_RULE_ID,
+          version: 3,
+          name: 'New test prebuilt rule',
+        });
+
+        setUpRuleUpgrades({
+          currentRuleAssets: [PREBUILT_RULE_ASSET],
+          rulePatches: [],
+          newRuleAssets: [NEW_PREBUILT_RULE_ASSET],
+        });
+        visitRulesUpgradeTable();
+        openPrebuiltRuleUpgradeFlyoutFor(PREBUILT_RULE_ASSET['security-rule'].name);
+
+        toggleFieldAccordion('name');
+        switchFieldToEditMode('name');
+
+        cy.get(UPDATE_PREBUILT_RULE_BUTTON).should('be.disabled');
+
+        saveFieldValue('name');
+
+        cy.get(UPDATE_PREBUILT_RULE_BUTTON).should('be.enabled');
+      });
+
+      it('is disabled when num of conflicts >= 1 and num fields in edit mode >= 1', () => {
+        const PREBUILT_RULE_ID = 'test-prebuilt-rule';
+        const PREBUILT_RULE_ASSET = createRuleAssetSavedObject({
+          rule_id: PREBUILT_RULE_ID,
+          version: 2,
+          name: 'Test prebuilt rule',
+          tags: ['tag-a'],
+        });
+        const NEW_PREBUILT_RULE_ASSET = createRuleAssetSavedObject({
+          rule_id: PREBUILT_RULE_ID,
+          version: 3,
+          name: 'New test prebuilt rule',
+          tags: ['tag-b'],
+        });
+
+        setUpRuleUpgrades({
+          currentRuleAssets: [PREBUILT_RULE_ASSET],
+          rulePatches: [{ rule_id: PREBUILT_RULE_ID, name: 'Customized prebuilt rule' }],
+          newRuleAssets: [NEW_PREBUILT_RULE_ASSET],
+        });
+        visitRulesUpgradeTable();
+        openPrebuiltRuleUpgradeFlyoutFor('Customized prebuilt rule');
+
+        toggleFieldAccordion('tags');
+        switchFieldToEditMode('tags');
+
+        cy.get(UPDATE_PREBUILT_RULE_BUTTON).should('be.disabled');
+
+        // Resolve name field conflicts
+        saveFieldValue('name');
+
+        // Make sure the "Update rule" button is still disabled
+        cy.get(UPDATE_PREBUILT_RULE_BUTTON).should('be.disabled');
+
+        // Cancel tags field value unchanged
+        cancelFieldValue('tags');
+
+        cy.get(UPDATE_PREBUILT_RULE_BUTTON).should('be.enabled');
+      });
+    });
+  }
+);
+
+const commonProperties: Partial<PrebuiltRuleAsset> = {
```

Comment:
Many of these reused properties fields and specific rule type assets seem to be shared between this file and `install_with_preview.cy.ts`, they could probably be factored out and stored in a mocks file either in this folder or in a higher, shared folder.

Reply:
Yes, you are right. I considered an option to share these assets between install and upgrade workflows but I'm not sure it's a good idea since it reduces test data locality and increases tests fragility. There is a chance that a little change in that data may cause unexpected side effects. So I'd defer sharing these assets until we have time to handle prebuilt rules installation workflow.